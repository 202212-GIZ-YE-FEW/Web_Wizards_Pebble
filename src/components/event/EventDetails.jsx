import { AiFillClockCircle } from "react-icons/ai";
import { MdLocationPin } from "react-icons/md";

import EventDate from "@/components/events/EventCard/EventDate.jsx";

import SingleDetail from "./SingleDetail.jsx";
const EventDetails = (props) => {
    const eventData = props?.eventData;
    return (
        <div className='px-0 xs:px-12 flex xs:flex-col sm:flex-col lg:flex-wrap md:flex-wrap'>
            <SingleDetail
                text={eventData?.location + " ,Yemen"}
                icon={<MdLocationPin size={30} />}
            />
            {/* <SingleDetail
                text={eventData?.address}
                icon={<MdLocationPin size={30} />}
            /> */}

            <EventDate
                eventDate={eventData?.date}
                icon={<AiFillClockCircle size={30} />}
            />
            <div className='flex -space-x-3 overflow-hidden'>
                {eventData?.attendees?.slice(0, 3).map((attende, index) => (
                    <div
                        key={index}
                        className=' h-8 w-8 rounded-full bg-gray-900 flex items-center justify-center text-white text-xs'
                    >
                        <span>{attende?.charAt(0)}</span>
                    </div>
                ))}
                <div className='px-2'>
                    <SingleDetail text='+12 Attendees' />
                </div>
            </div>
            <SingleDetail
                text={`Organized by ${eventData?.organizer}`}
                icon={
                    <div className='h-8 w-8 rounded-full bg-gray-900 flex items-center justify-center text-white text-xs'>
                        <span>{eventData?.organizer?.charAt(0)}</span>
                    </div>
                }
            />
        </div>
    );
};

export default EventDetails;
