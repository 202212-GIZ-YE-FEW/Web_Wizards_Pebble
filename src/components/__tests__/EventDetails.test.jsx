import renderer from "react-test-renderer";

import EventDetails from "../event/EventDetails";

it("renders correctly", () => {
    const eventData = {
        title: "Meeting Brainstorm Energy CIeccan lzmir",
        location: "Izmir, Turkey",
        date: "ahhallsei 26 Augest Kapisi",
        organizer: "Sufyan",
        attendees: ["Sufyan", "Adeeb", "Abdullah", "Azzam", "Hamad"],
    };
    const tree = renderer
        .create(<EventDetails eventData={eventData} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
