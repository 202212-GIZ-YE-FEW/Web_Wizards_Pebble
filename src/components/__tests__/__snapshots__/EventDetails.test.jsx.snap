// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-0 xs:px-12 flex xs:flex-col sm:flex-col lg:flex-wrap md:flex-wrap"
>
  <div
    className="lg:w-full md:w-1/2 sm:w-full mb-4 xs:w-full"
  >
    <h3
      className="font-semibold lg:text-lg text-md"
    >
      <span
        className="mr-2 inline-block"
      >
        <svg
          fill="currentColor"
          height={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 1.74.5 3.37 1.41 4.84.95 1.54 2.2 2.86 3.16 4.4.47.75.81 1.45 1.17 2.26.26.55.47 1.5 1.26 1.5s1-.95 1.25-1.5c.37-.81.7-1.51 1.17-2.26.96-1.53 2.21-2.85 3.16-4.4C18.5 12.37 19 10.74 19 9c0-3.87-3.13-7-7-7zm0 9.75a2.5 2.5 0 010-5 2.5 2.5 0 010 5z"
          />
        </svg>
      </span>
      <span
        className="inline-block"
      >
        Izmir, Turkey
      </span>
    </h3>
  </div>
  <div
    className="lg:w-full md:w-1/2 sm:w-full mb-4 xs:w-full"
  >
    <h3
      className="font-semibold lg:text-lg text-md"
    >
      <span
        className="mr-2 inline-block"
      >
        <svg
          fill="currentColor"
          height={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm176.5 585.7l-28.6 39a7.99 7.99 0 0 1-11.2 1.7L483.3 569.8a7.92 7.92 0 0 1-3.3-6.5V288c0-4.4 3.6-8 8-8h48.1c4.4 0 8 3.6 8 8v247.5l142.6 103.1c3.6 2.5 4.4 7.5 1.8 11.1z"
          />
        </svg>
      </span>
      <span
        className="inline-block"
      >
        ahhallsei 26 Augest Kapisi
      </span>
    </h3>
  </div>
  <div
    className="flex -space-x-3 overflow-hidden"
  >
    <div
      className=" h-8 w-8 rounded-full bg-gray-900 flex items-center justify-center text-white text-xs"
    >
      <span>
        S
      </span>
    </div>
    <div
      className=" h-8 w-8 rounded-full bg-gray-900 flex items-center justify-center text-white text-xs"
    >
      <span>
        A
      </span>
    </div>
    <div
      className=" h-8 w-8 rounded-full bg-gray-900 flex items-center justify-center text-white text-xs"
    >
      <span>
        A
      </span>
    </div>
    <div
      className="px-2"
    >
      <div
        className="lg:w-full md:w-1/2 sm:w-full mb-4 xs:w-full"
      >
        <h3
          className="font-semibold lg:text-lg text-md"
        >
          <span
            className="mr-2 inline-block"
          />
          <span
            className="inline-block"
          >
            +12 Attendees
          </span>
        </h3>
      </div>
    </div>
  </div>
  <div
    className="lg:w-full md:w-1/2 sm:w-full mb-4 xs:w-full"
  >
    <h3
      className="font-semibold lg:text-lg text-md"
    >
      <span
        className="mr-2 inline-block"
      >
        <div
          className="h-8 w-8 rounded-full bg-gray-900 flex items-center justify-center text-white text-xs"
        >
          <span>
            S
          </span>
        </div>
      </span>
      <span
        className="inline-block"
      >
        Organized by Sufyan
      </span>
    </h3>
  </div>
</div>
`;
